lightning/src/chain/chainmonitor.rs

```diff
@@ -656,6 +661,27 @@ where C::Target: chain::Filter,
 			}
 		}
 	}
+
+	/// Archives fully resolved channel monitors by calling [`Persist::archive_persisted_channel`].
+	///
+	/// This is useful for pruning fully resolved monitors from the monitor set and primary
+	/// storage so they are not reloaded on every new new block connection.
+	///
+	/// Depending on the `[Persist::archive_persisted_channel]` the monitor data could be moved
+	/// to an archive location or/and removed  entirely.
+	pub fn archive_fully_resolved_channel_monitors(&self, to_archive: Vec<OutPoint>) {
```

Comment:
Wouldn't that introduce a good amount of redundant calls to `is_fully_resolved`? the user could always pass the full list but they also can pass a subset. maybe it could be an Option and if None then run on full list?